### ISSUE TITLE
DataTable: unselect not clearing selection in PFX

### ISSUE THREAD

**Issue by matsandreassen** [primefaces/primefaces#7128]
**Describe the defect**
Testing out PFX. While following the migration guide, I noticed a problem with Datatable when performing a _unselect_.

In my datatables, I noticed that _selection_ was not updated even though the rowUnselect event listener was called properly. Also, when I clear the selection from the model in my backing bean and update/re-render the datatable, the row would remain selected.

**Environment:**
 - PF Version: _10.0.0_
 - JSF + version: Mojarra 2.2.17
 - Affected browsers: _ALL_

**To Reproduce**
1. Go to the Showcase
2. Navigate to Datatable - Selection
3. Scroll down to 'Single with Row Click'
4. Try to click select a row, and press 'View'. Afterwards, CTRL click the row, thereby unselecting it. When pressing 'View' again, it will still display the previously selected item.

**Expected behavior**
I expected the selection property of the datatable to be set to null, like in previous versions of PF.

**Sample code**
I'm not attaching any code since it can be reproduced in the showcase itself. I hope that's ok.

**Comment by melloware** [comment]
I have verified this issue and will investigate.

**Comment by melloware** [comment]
OK I have verified the JS code is definitely unselecting the item.
**Selected:**
![image](https://user-images.githubusercontent.com/4399574/111321892-c0763a00-863e-11eb-8952-4e374e19ca39.png)

**Unselected:**
![image](https://user-images.githubusercontent.com/4399574/111322325-311d5680-863f-11eb-88a3-10743e4152e8.png)

So I think that means the SelectionFeature or something is not clearing out the `view.selection` of the backing bean.

**Comment by matsandreassen** [comment]
Fantastic! Looking forward to continue my PFX migration.

**Comment by silvio3punkt5** [comment]
**Description**
There is still an issue that unselecting - by clearing the list in the bean via action method - does not clear the selection in the UI.

**To Reproduce / Sample Code:**
Using the latest showcase example in DataTable -> Selection -> Checkbox.

Add Clear-Button in selection.xhtml:174:
`
                    <f:facet name="footer">
                        <p:commandButton process="checkboxDT" update=":form:checkboxDT" value="Clear" action="#{dtSelectionView.clearProducts}" />
                    </f:facet>
`

Add Clear-Action in org.primefaces.showcase.view.data.datatable.SelectionView:
`
    public void clearProducts(){
        selectedProducts.clear();
    }
`

**Expected behavior**
The rows should be unselected. Adding the same sample code on v8.0 the rows are properly unselected.

**Comment by melloware** [comment]
@silvio3punkt5 have you tried it with 11.0.0-SNAPSHOT that has this PR applied?

**Comment by silvio3punkt5** [comment]
@melloware yes, I've used 11.0.0-SNAPSHOT including the commit from above

**Comment by Rapster** [comment]
I'm having a look

**Comment by christophs78** [comment]
we should add a integration-test for this :-)

**Comment by matsandreassen** [comment]
Hello again. I just updated to 10.0.4 and I still can't unselect using backing bean like I described in OP. Also confirmed by silvio3punkt5 further down.

I forked so that it can be verified in https://github.com/matsandreassen/primefaces-test

**Comment by christophs78** [comment]
We also need to recheck/enhance https://github.com/primefaces/primefaces/blob/master/primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/datatable/DataTable004Test.java.

**Comment by christophs78** [comment]
@matsandreassen Can you reproduce this issue with 11.0.0-SNAPSHOT? (IMO it works fine.)
And the integration test IMO does the right checks.
Maybe PrimeTek made some kind of mistake when they picked this fix for 10.0.1? (elite-branches are not public so we can´t check this.)
(I´m talking about single selection mode.)

@silvio3punkt5 - You talked about Checkbox. So you mean a multiselect-usecase? Maybe this is another issue.

**Comment by christophs78** [comment]
https://github.com/primefaces/primefaces/pull/7688 enhances our existing integrationtests. 11.0.0-SNAPSHOT seems to work fine with multiselect/checkbox.
@mertsincan, @sqores: Maybe something is still broken for 10.0.X - elite release?

**Comment by matsandreassen** [comment]
> @matsandreassen Can you reproduce this issue with 11.0.0-SNAPSHOT? (IMO it works fine.)

I would if I find out how to replace 10.0.4 with the snapshot artifact. I tried inputting

```xml
<dependency>
            <groupId>com.github.primefaces</groupId>
            <artifactId>primefaces</artifactId>
            <version>master-SNAPSHOT</version>
</dependency>
```

into primefaces-test but it's not resolving properly.

**Comment by melloware** [comment]
When using the master SNAPSHOT from jitpack it can take a while to build and publish before it resolves. Go here and click on the "Builds" tab you will see Master SNAPSHOT is still building..

https://jitpack.io/#primefaces/primefaces

**Comment by matsandreassen** [comment]
The snapshot works as desired! The hypothesis seems to check out, something seems to have gone wrong when merging to elite.

**Comment by matsandreassen** [comment]
@melloware Is something happening with this? Can I do anything to help ensure this gets fixed in elite?

**Comment by melloware** [comment]
@matsandreassen I don't work for PrimeTek so its up to @sqores to put into Elite.

**Comment by matsandreassen** [comment]
Hello @sqores: This issue already has a 10.0.1 ELITE label but it was definitely not fixed in that release. Should not this issue be re-opened for 10.0.6?

**Comment by buunsal** [comment]
Hello @matsandreassen I'm examining. I will do the necessary fixes for 10.0.6.

**Comment by buunsal** [comment]
@matsandreassen Could you please try again with 10.0.5? I couldn't reproduce your issue when I tested

https://user-images.githubusercontent.com/45278820/129567666-f7eec165-70d6-4ebf-b0fa-f8bc13c846c0.mov

**Comment by matsandreassen** [comment]
Not so fast. ;)

I originally described two facets to this problem. The first is the one you demonstrate in your video to be solved. 

The second one is not solved. If you clear the selection in the backing bean and update the table, the UI still indicates the row to be selected. I provided a primefaces-test fork where it's easily reproduced: https://github.com/matsandreassen/primefaces-test/tree/unselect

The last thing we discovered is that it works with 11-SNAPSHOT.

**Comment by melloware** [comment]
@matsandreassen that is because that was a completely different issue and reported and fixed here for 11.0:  https://github.com/primefaces/primefaces/issues/7391

The fix is right here in SelectionFeature.java: https://github.com/primefaces/primefaces/commit/ebee2cc303dea786761a0b15a277529e94434ee7#diff-5ca105b8da2e2839895b44587227b2d3440201b71dbd9c13f2ab15a6cb3c22b2

**Comment by matsandreassen** [comment]
I assumed this was the issue since I reported it myself back in March. Someone else reported it AND it was fixed while I was on summer vacation. That's great! 

@sqores Please add #7391 for elite release. 👍
